src/screens/Ticketing/Tickets/use-recent-tickets.ts

```diff
@@ -0,0 +1,200 @@
+        loading: false,
+        error: true,
+      };
+    case 'SUCCESS':
+      return {
+        ...prevState,
+        loading: false,
+        error: false,
+        recentFareContracts: action.data,
+      };
+  }
+};
+
+const mapUsers = (
+  users: {[user_profile: string]: string},
+  userProfiles: UserProfile[],
+) =>
+  Object.entries(users)
+    .map(([profileId, count]) => ({
+      ...findReferenceDataById(userProfiles, profileId),
+      count: parseInt(count),
+    }))
+    .filter((u): u is UserProfileWithCount => 'id' in u)
+    .sort(
+      // Sort by user profile order in remote config
+      (u1, u2) =>
+        userProfiles.findIndex((profile) => profile.id === u1.id) -
+        userProfiles.findIndex((profile) => profile.id === u2.id),
+    );
+
+const mapRecentFareContractToRecentTicket = (
+  recentFareContract: RecentFareContract,
+  {
+    tariff_zones: tariffZones,
+    user_profiles: userProfiles,
+    preassigned_fare_products: preassignedFareProducts,
+  }: RemoteConfigContextState,
+) => ({
+  preassignedFareProduct: findReferenceDataById(
+    preassignedFareProducts,
+    recentFareContract.products[0],
+  ),
+  fromTariffZone: findReferenceDataById(
+    tariffZones,
+    recentFareContract.zones[0],
+  ),
+  toTariffZone: findReferenceDataById(
+    tariffZones,
+    recentFareContract.zones.slice(-1)[0],
+  ),
+  userProfilesWithCount: mapUsers(recentFareContract.users, userProfiles),
+});
+
+const isUsersEqual = (
+  users1: UserProfileWithCount[],
+  users2: UserProfileWithCount[],
+) => {
+  if (users1.length !== users2.length) return false;
+
+  return users1.reduce(
+    (isEqual, user1) =>
+      isEqual &&
+      users2.some(
+        (user2) => user1.id === user2.id && user1.count === user2.count,
+      ),
+    true,
+  );
+};
+
+const isTicketsEqual = (t1: RecentTicket, t2: RecentTicket) =>
+  t1.preassignedFareProduct.id === t2.preassignedFareProduct.id &&
+  t1.fromTariffZone.id === t2.fromTariffZone.id &&
+  t1.toTariffZone.id === t2.toTariffZone.id &&
+  isUsersEqual(t1.userProfilesWithCount, t2.userProfilesWithCount);
+
+const containsTicket = (
+  existingTickets: RecentTicket[],
+  currentTicket: RecentTicket,
+) =>
+  existingTickets.some((existing) => isTicketsEqual(existing, currentTicket));
+
+/**
+ * This method:
+ * - Maps from the api RecentFareContract format to the RecentTicket format
+ * - Filter out tickets where the product or tariff zones were not found in our
+ *   reference data, or none of the user profiles were found in our reference
+ *   data
+ * - Sort descending by created date
+ * - Remove similar tickets, which means tickets with same product, same zones
+ *   and same travellers
+ * - Return only the first three tickets, which will be the three most recent
+ *   tickets
+ */
+const mapToLastThreeUniqueTickets = (
+  recentFareContracts: RecentFareContract[],
+  remoteConfigState: RemoteConfigContextState,
+): RecentTicket[] => {
+  return recentFareContracts
```

Comment:
What is typically the size of this?

Reply:
I'll look into it.

Reply:
All recent orders from Entur is returned. Martin is not sure if they expire or not at some point. But, as of now we hope to get Firestore in place before next production release. When Firestore is used we will not continuously retrieve all elements, but just the diff that has changed. I will also argue for that the data should already be sorted when stored in Firestore.